### ISSUE TITLE
Preparação dos dados

### ISSUE THREAD

**Issue by omadson** [atlantico-academy/risk-prediction#10]
A equipe deve criar todo o código necessário (podendo ser no notebook `notebooks/02-comparative_analysis.ipynb`) para a preparação dos dados. O notebook deve conter os seguintes tratamentos:
 - [x] Tratamento de dados discrepantes
 - [x] Tratamento de dados faltantes
 - [x] Codificação de variáveis
 - [x] Seleção de variáveis
 - [x] Normalização

**Comment by marcelloale** [comment]
Para 'Seleção de variáveis' eu testei com o [sklearn.ensemble.ExtraTreesClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.ExtraTreesClassifier.html#sklearn.ensemble.ExtraTreesClassifier) queria saber se é valido
```
#df.drop('ID',axis=1, inplace=True)
array = df.values
aX = array[:,0:22]
ay = array[:,23]

#Essa classe implementa um meta estimador que ajusta várias árvores de decisão aleatórias (também conhecidas como extra-árvores) em várias subamostras do conjunto de dados e usa a média para melhorar a precisão preditiva e controlar o ajuste excessivo.
select = ExtraTreesClassifier()
select.fit(aX, ay)

#Resultados
print(df.columns[0:23])
print(select.feature_importances_)
```
Index(['LIMIT_BAL', 'SEX', 'EDUCATION', 'MARRIAGE', 'AGE', 'PAY_0', 'PAY_2',
       'PAY_3', 'PAY_4', 'PAY_5', 'PAY_6', 'BILL_AMT1', 'BILL_AMT2',
       'BILL_AMT3', 'BILL_AMT4', 'BILL_AMT5', 'BILL_AMT6', 'PAY_AMT1',
       'PAY_AMT2', 'PAY_AMT3', 'PAY_AMT4', 'PAY_AMT5', 'PAY_AMT6'],
      dtype='object')
[0.06968273 0.01102185 0.03399222 0.02279865 0.07106435 0.1016439
 0.04285968 0.03408487 0.0307821  0.03064965 0.03086521 0.05388978
 0.04999096 0.04821095 0.04766745 0.04666712 0.04769019 0.0465657
 0.04515631 0.04470961 0.04409334 0.04591338]
```
nomes=['LIMIT_BAL', 'SEX', 'EDUCATION', 'MARRIAGE', 'AGE', 'PAY_0', 'PAY_2',
       'PAY_3', 'PAY_4', 'PAY_5', 'PAY_6', 'BILL_AMT1', 'BILL_AMT2',
       'BILL_AMT3', 'BILL_AMT4', 'BILL_AMT5', 'BILL_AMT6', 'PAY_AMT1',
       'PAY_AMT2', 'PAY_AMT3', 'PAY_AMT4', 'PAY_AMT5', 'PAY_AMT6']
porcent=[0.06941332,0.01212699,0.03416065,0.02216255,0.07089877,0.09761976,
         0.0442349,0.03986343,0.0275776,0.0298341,0.03215941,0.05357525,
         0.04943215,0.04813462,0.04764368,0.04646116,0.04763427,0.04723214,
         0.04513238,0.04485132,0.04379029,0.04606124]
dict = {}
for i,j in zip(nomes, porcent):
    dict[i] = j
for i in sorted(dict, key = dict.get, reverse=True):
    print(i, dict[i])
```
PAY_0 0.09761976
AGE 0.07089877
LIMIT_BAL 0.06941332
BILL_AMT1 0.05357525
BILL_AMT2 0.04943215
BILL_AMT3 0.04813462
BILL_AMT4 0.04764368
BILL_AMT6 0.04763427
PAY_AMT1 0.04723214
BILL_AMT5 0.04646116
PAY_AMT5 0.04606124
PAY_AMT2 0.04513238
PAY_AMT3 0.04485132
PAY_2 0.0442349
PAY_AMT4 0.04379029
PAY_3 0.03986343
EDUCATION 0.03416065
PAY_6 0.03215941
PAY_5 0.0298341
PAY_4 0.0275776
MARRIAGE 0.02216255
SEX 0.01212699

**Comment by marcelloale** [comment]
Utilizei as bibliotecas 
`from sklearn.feature_selection import SelectKBest, mutual_info_classif`
Adicionei no Pipeline a seleção das variaveis continuas por filtragem utilizando o SelectKBest utilizando como função de score o mutual_info_classif selecionando as 10 (50%) melhores das 20 variaveis continuas do dataset
`('feature_selection', SelectKBest(score_func=mutual_info_classif, k=10))`
Nas variaveis nominais não foi aplicada por ser apenas 3 variaveis
`# não aplicar seleção apenas 3 variaveis nominais SEX, EDUCATION e MARRIAGE `
